src/IO/S3/Client.cpp

```diff
@@ -846,7 +837,8 @@ std::unique_ptr<S3::Client> ClientFactory::create( // NOLINT
             std::move(credentials),
             credentials_configuration);
 
-    client_configuration.retryStrategy = std::make_shared<Client::RetryStrategy>(std::move(client_configuration.retryStrategy));
+    client_configuration.retryStrategy = std::make_shared<Client::RetryStrategy>(client_configuration.s3_retry_attempts);
```

Comment:
I think `MaxDelayMs` should also be changeable by setting here.

Reply:
I will add a dedicated setting for it when I find a reason to customize it from node to node.